src/include/ipc/probe.h

```diff
@@ -0,0 +1,134 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ *         Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+/**
+ * \file include/ipc/probe.h
+ * \brief Probe IPC definitions
+ * \author Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ * \author Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __IPC_PROBE_H__
+#define __IPC_PROBE_H__
+
+#include <ipc/header.h>
+#include <sof/bit.h>
+#include <stdint.h>
+
+#define PROBE_PURPOSE_EXTRACTION	0x1
+#define PROBE_PURPOSE_INJECTION		0x2
+
+#define PROBE_EXTRACT_SYNC_WORD		0xBABEBEBA
+
+#define PROBE_SHIFT_FMT_TYPE		31
+#define PROBE_SHIFT_STANDARD_TYPE	27
+#define PROBE_SHIFT_AUDIO_FMT		23
+#define PROBE_SHIFT_SAMPLE_RATE		19
+#define PROBE_SHIFT_NB_CHANNELS		14
+#define PROBE_SHIFT_SAMPLE_SIZE		12
+#define PROBE_SHIFT_CONTAINER_SIZE	10
+#define PROBE_SHIFT_SAMPLE_FMT		9
+#define PROBE_SHIFT_SAMPLE_END		8
+#define PROBE_SHIFT_INTERLEAVING_ST	7
+
+#define PROBE_MASK_FMT_TYPE		MASK(31, 31)
+#define PROBE_MASK_STANDARD_TYPE	MASK(30, 27)
+#define PROBE_MASK_AUDIO_FMT		MASK(26, 23)
+#define PROBE_MASK_SAMPLE_RATE		MASK(22, 19)
+#define PROBE_MASK_NB_CHANNELS		MASK(18, 14)
+#define PROBE_MASK_SAMPLE_SIZE		MASK(13, 12)
+#define PROBE_MASK_CONTAINER_SIZE	MASK(11, 10)
+#define PROBE_MASK_SAMPLE_FMT		MASK(9, 9)
+#define PROBE_MASK_SAMPLE_END		MASK(8, 8)
+#define PROBE_MASK_INTERLEAVING_ST	MASK(7, 7)
+
+/**
+ * Header for data packets sent via compressed PCM from extraction probes
+ */
+struct probe_data_packet {
+	uint32_t sync_word;		/**< PROBE_EXTRACT_SYNC_WORD */
+	uint32_t buffer_id;		/**< Buffer ID from which data was extracted */
+	uint32_t format;		/**< Encoded data format */
+	uint32_t timestamp_low;		/**< Low 32 bits of timestamp in us */
+	uint32_t timestamp_high;	/**< High 32 bits of timestamp in us */
+	uint32_t checksum;		/**< CRC32 of header and payload */
+	uint32_t data_size_bytes;	/**< Size of following audio data */
+	uint32_t data[];		/**< Audio data extracted from buffer */
+} __attribute__((packed));
+
+/**
+ * Description of probe dma
+ */
+struct probe_dma {
+	uint32_t stream_tag;		/**< Stream tag associated with this DMA */
+	uint32_t dma_buffer_size;	/**< Size of buffer associated with this DMA */
```

Comment:
buffer on the host size (i.e. a ring buffer with multiple periods/segments) or the DSP side?

Reply:
Depending on context, when sent from host - host size, when sent as reply to info - dsp size.
We're planning of having fixed size buffers on DSP side, similarly to how traces have it.

Reply:
sorry, I don't get the answer. why would the firmware care about the buffer size on the host side?

Reply:
I've just checked our development branch for this particular variable usage, and it seems, that we only hold the value sent from kernel and then report it back. I haven't found any actual use of these values. I guess that it was inherited from cAVS and never used afterwards.

Reply:
hum, interesting. In that case, what is the point of the INFO IPC for probe_dma?
At least for HDaudio platforms, the stream_tag would be allocated by the Linux driver, so there wouldn't really be a need to query anything from the firmware that the driver doesnt' already know.

The other question this raises is whether this probe mechanism would work on older platforms with a GP-DMA?

Reply:
@plbossart this is how it works in cAVS and also there was a request from kernel-side dev to make it possible to ask fw for dma info.

Reply:
so we still don't know what the purpose of this field is... The information is provided to the firmware by the driver and re-read by the driver, that seems very odd to me.